pkg/apachepulsar/pulsar_source.go

```diff
@@ -55,8 +55,15 @@ func (ps *PulsarSource) Read(_ context.Context, readRequest sourcesdk.ReadReques
 		default:
 			ps.lock.Lock()
 			msg, err := ps.consumer.Receive(ctx)
+			if msg == nil {
+				log.Println("received empty message")
+				ps.lock.Unlock()
```

Comment:
You can use `defer` so that you just need to write `ps.lock.Unlock()` once, which is right after `ps.lock.Lock()`

Reply:
Thanks for contributing!

Reply:
unit tests failed - https://github.com/numaproj-contrib/apache-pulsar-source-go/actions/runs/8392192965/job/22986664593?pr=2

Could you please take a look? Thanks!